### PR TITLE
feat: publish raw_twist regardless of velocity_valid

### PR DESCRIPTION
**Pull request by tjsos**
Added another topic `dvl/raw_twist` that publishes twist msg continuously even when veloctiy_valid is False.

### CODE REVIEW

File: src/waterlinked_dvl/waterlinked_dvl_tcp.cpp [hunk 1]
```diff
@@ -190,6 +192,18 @@ void WaterlinkedDvlTcp::f_parse_json_v2(Json::Value root) {
...
                 m_twist_publisher.publish(twist_msg);
             }
+            //Raw Twist
+            geometry_msgs::TwistWithCovarianceStamped twist_msg;
+            twist_msg.header = msg.header;
+            twist_msg.twist.twist.linear.x = msg.report.vx;
+            twist_msg.twist.twist.linear.y = msg.report.vy;
+            twist_msg.twist.twist.linear.z = msg.report.vz;
+
+            if (m_velocity_covariance.size() == twist_msg.twist.covariance.size()) {
```
Comment (mzhouURI) [inline src/waterlinked_dvl/waterlinked_dvl_tcp.cpp]: covariance should from the transducer report not a user defined value. the mapping of the values are wrong.

File: src/waterlinked_dvl/waterlinked_dvl_tcp.cpp [hunk 2]
```diff
@@ -303,6 +317,17 @@ void WaterlinkedDvlTcp::f_parse_json_v3(Json::Value root)
...
```
Comment (mzhouURI) [inline src/waterlinked_dvl/waterlinked_dvl_tcp.cpp]: covariance source is wrong. it should from the transducer report

File: src/waterlinked_dvl/waterlinked_dvl_tcp.cpp [hunk 3]
```diff
@@ -190,6 +192,18 @@ void WaterlinkedDvlTcp::f_parse_json_v2(Json::Value root) {
                 }
                 m_twist_publisher.publish(twist_msg);
             }
+            //Raw Twist
+            geometry_msgs::TwistWithCovarianceStamped twist_msg;
```
Comment (mzhouURI) [inline src/waterlinked_dvl/waterlinked_dvl_tcp.cpp]: raw twist needs to be moved before the velocity validation if statement. for both v2 and v3.

**Review by mzhouURI — CHANGES_REQUESTED** [review]

File: src/waterlinked_dvl/waterlinked_dvl_tcp.cpp (hunk 2 above)
Comment (tjsos) [reply src/waterlinked_dvl/waterlinked_dvl_tcp.cpp]: Fixed. Now mapping covariance values from transducer report to the v_dot submatrix in the covariance matrix.

File: src/waterlinked_dvl/waterlinked_dvl_tcp.cpp (hunk 3 above)
Comment (tjsos) [reply src/waterlinked_dvl/waterlinked_dvl_tcp.cpp]: Moved.

File: src/waterlinked_dvl/waterlinked_dvl_tcp.cpp (hunk 1 above)
Comment (tjsos) [reply src/waterlinked_dvl/waterlinked_dvl_tcp.cpp]: json_v2 format do not have covariance matrix.
dvl/twist_raw can only then be derived from json_v3 format.

**Comment by tjsos** [comment]
json_v2 format of the transducer report doesn't contain covariance values. 
dvl/twist_raw msg can only then be derived from json_v3 format. Covariance values are mapped to the v_dot submatrix in the covariance matrix. The message also published before the velocity_valid check.

**Pull request merged**